include/revng/ADT/UpcastablePointer.h

```diff
@@ -198,14 +204,19 @@ class UpcastablePointer {
       Other.upcast([&](auto &OtherUpcasted) {
         using ThisType = std::remove_cvref_t<decltype(Upcasted)>;
         using OtherType = std::remove_cvref_t<decltype(OtherUpcasted)>;
-        if constexpr (std::is_same_v<ThisType, OtherType>) {
+        if constexpr (std::is_same_v<ThisType, OtherType>)
           Result = Upcasted == OtherUpcasted;
-        }
       });
     });
     return Result;
   }
 
+  template<UpcastablePointerLike PointerLike>
+    requires(std::three_way_comparable_with<T, Dereferenced<PointerLike>>)
+  auto operator<=>(const PointerLike &Other) const {
+    return *Pointer <=> *Other;
+  }
```

Comment:
Can you elaborate what the "(squash) revamp model type comparisons" does and why?